dpctl/tensor/_copy_utils.py

```diff
@@ -924,6 +924,8 @@ def _place_impl(ary, ary_mask, vals, axis=0):
     else:
         rhs = dpt.astype(vals, ary.dtype)
     rhs = dpt.broadcast_to(rhs, expected_vals_shape)
+    if mask_nelems == 0:
+        return
```

Comment:
Is it possible to write this check earlier (i.e., line 910) as a very small optimization?

Reply:
I was considering that, but that the check that value array can be broadcast to the desired shape would be skipped. I think error checking should not be different between generic case and this special case.